game-core/src/test/java/games/strategy/triplea/delegate/battle/FakeBattleState.java

```diff
@@ -96,12 +94,34 @@
   }
 
   @Override
-  public BattleRound getBattleRoundState() {
-    return BattleRound.of(battleRound, maxBattleRounds);
+  public BattleStatus getStatus() {
+    return BattleStatus.of(battleRound, maxBattleRounds, over, amphibious, headless);
   }
 
   @Override
-  public Collection<Unit> getUnits(final Side... sides) {
+  public GamePlayer getPlayer(final Side side) {
+    return side == OFFENSE ? attacker : defender;
+  }
+
+  @Override
+  public Collection<Unit> getUnits(final UnitsStatus status, final Side... sides) {
```

Comment:
The exact implementation in the fake as we have in the real is concerning. At some point this is not even a fake but just another battle-state implementation.  I don't know if there is a good solution for this problem that would fit within the scope of this PR. However, let's keep an eye on this as it seems to be a growing problem for the fake battle state.

Reply:
Yeah, it is turning into a real implementation.  I've started to think that it might be better to pull it out as a real implementation and change `MustFightBattle` to hold a transient instance of it.  Then instead of passing `MustFightBattle` around as a `BattleState`, it would instead pass this implementation.